### PR TITLE
683 join keys@main

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Part of https://github.com/insightsengineering/teal/pull/697

### CODE REVIEW

File: R/data_extract_module.R [hunk 1]
```diff
@@ -395,31 +398,25 @@ data_extract_srv.FilteredData <- function(id, datasets, data_extract_spec, ...)
...
-#' @param join_keys (named `list` or `NULL`) of keys per dataset in `datasets`
+#' @param join_keys (`JoinKeys` or `NULL`) of keys per dataset in `datasets`
 #' @export
 data_extract_srv.list <- function(id, datasets, data_extract_spec, join_keys = NULL, ...) {
   checkmate::assert_list(datasets, types = c("reactive", "data.frame"), names = "named")
-  if (length(join_keys) == 0) join_keys <- NULL
-  checkmate::assert_list(join_keys, names = "named", null.ok = TRUE)
-  checkmate::assert(
-    .var.name = "join_keys",
-    checkmate::check_names(names(join_keys), subset.of = names(datasets)),
```
Comment (nikolas-burkoff) [inline R/data_extract_module.R:420]: @mhallal1 I've removed this check here and in the other two places (and the corresponding tests) as it no longer works and also I don't think it quite makes sense as the while join keys object can have entries for other dataset if we want. And hopefully `validate_keys_sufficient` is good enough when it's called

File: R/get_merge_call.R [hunk 2]
```diff
@@ -118,11 +116,10 @@ get_merge_call <- function(selector_list,
 #' @inheritParams get_merge_call
 #' @return list of key pairs between all datasets
 #' @keywords internal
-get_merge_key_grid <- function(selector_list, join_keys = list()) {
+get_merge_key_grid <- function(selector_list, join_keys = teal.data::join_keys()) {
   logger::log_trace(
     paste(
-      "get_merge_key_grid called with: { paste(names(selector_list), collapse = ', ') } selectors;",
-      "{ length(join_keys) } key pairs."
+      "get_merge_key_grid called with: { paste(names(selector_list), collapse = ', ') } selectors;"
```
Comment (mhallal1) [inline R/get_merge_call.R:122]:
```suggestion
      "get_merge_key_grid called with: { paste(names(selector_list), collapse = ', ') } selectors."
```

File: R/get_merge_call.R [hunk 3]
```diff
@@ -118,11 +116,10 @@ get_merge_call <- function(selector_list,
 #' @inheritParams get_merge_call
 #' @return list of key pairs between all datasets
 #' @keywords internal
-get_merge_key_grid <- function(selector_list, join_keys = list()) {
+get_merge_key_grid <- function(selector_list, join_keys = teal.data::join_keys()) {
   logger::log_trace(
     paste(
```
Comment (mhallal1) [inline R/get_merge_call.R:121]: do we still require the paste here?

Comment (nikolas-burkoff) [reply R/get_merge_call.R]: nope

File: R/data_extract_module.R [hunk 4]
```diff
@@ -342,7 +345,7 @@ check_data_extract_spec_react <- function(datasets, data_extract) {
 #' # Using reactive list of data.frames
 #' data_list <- list(ADSL = reactive(ADSL))
 #'
-#' key_list <- list(ADSL = c("STUDYID", "USUBJID"))
+#' key_list <- teal.data::join_keys(teal.data::join_key("ADSL", "ADSL", c("STUDYID", "USUBJID")))
```
[the remark this answers is not in the document]
Comment (nikolas-burkoff) [reply R/data_extract_module.R]: yup done

**Review by mhallal1 — APPROVED** [review]

**Pull request merged**